src/js/git/commands.js

```diff
@@ -674,7 +675,8 @@ var commandConfig = {
 
```

Comment:
the argument parsing in this function is pretty janky (and wrong in a lot of places) but worth adding on any arguments after the boolean flags here, otherwise this would work:
```
git rebase main --update-refs
```

but this would fail:
```
git rebase --update-refs main
```

since `main` would be stored in the options map

so maybe something like

```
generalArgs = generalArgs.concat(commandOptions['-p'] || commandOptions['--preserve-merges'] || []).concat(commandOptions['--update-refs'] || [])
```